apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -60,248 +73,192 @@ export function OAuth2Form() {
-      nonce: getURLParam('nonce'),
-    }
-    if (claimsString) {
-      authRequest.claims = JSON.parse(claimsString)
-    }
-    return authRequest
-  }
+  const { data: oidcRequestDescription, error: oidcRequestDescriptionError } =
+    SynapseQueries.useGetOAuth2RequestDescription(
+      oidcAuthorizationRequestFromSearchParams!,
+      {
+        enabled: Boolean(oidcAuthorizationRequestFromSearchParams),
+      },
+    )
 
-  //init
   useEffect(() => {
-    const getOauthClientInfo = () => {
-      if (!oauthClientInfo && !error) {
-        const code = getURLParam('code')
-        if (code) return // we're in the middle of a SSO, do not attempt to get OAuthClient info yet
+    if (oidcRequestDescription) {
+      sendGTagEvent('SynapseOAuthClientRequestDescriptionLoaded')
 
-        const clientId = getURLParam('client_id')
-        if (!clientId) {
-          onError(new Error('Synapse OAuth client_id is required'))
-          return
-        }
-        SynapseClient.getOAuth2Client(clientId)
-          .then((oauthClientInfo: OAuthClientPublic) => {
-            if (oauthClientInfo.verified) {
-              getOAuth2RequestDescription()
-            }
-            setOauthClientInfo(oauthClientInfo)
-          })
-          .catch(_err => {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-            onError(_err)
-          })
+      // if we were able to get the oidc request description, also check for params that this web app does not support
+      // sorry, we don't support JWT in the url query params today
+      // https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests
+      const requestObject = queryParams.get('request')
+      const requestUri = queryParams.get('request_uri')
+      if (requestObject) {
+        handleErrorRedirect(new OAuthClientError('request_not_supported'))
+      }
+      if (requestUri) {
+        handleErrorRedirect(new OAuthClientError('request_uri_not_supported'))
+      }
+      // sorry, we don't support registration (yet?)
+      // https://openid.net/specs/openid-connect-core-1_0.html#RegistrationParameter
+      const registration = queryParams.get('registration')
+      if (registration) {
+        handleErrorRedirect(new OAuthClientError('registration_not_supported'))
       }
     }
-
-    getOauthClientInfo()
-  }, [error, getOAuth2RequestDescription, oauthClientInfo, onError])
+  }, [oidcRequestDescription, queryParams])
 
   useEffect(() => {
-    if (redirectURL) {
-      window.location.replace(redirectURL)
+    if (oidcRequestDescriptionError) {
+      onError(oidcRequestDescriptionError)
     }
-  }, [redirectURL])
+  }, [oidcRequestDescriptionError, onError])
 
-  const isLoadingProfile =
-    !twoFactorAuthSSOErrorResponse &&
-    !error &&
-    profile &&
-    !isPreviousAuthCheckComplete
-  const isLoadingClientInfo =
-    !error && !oauthClientInfo && !oidcRequestDescription
-  const isRedirecting = redirectURL && oauthClientInfo
```

Comment:
`isRedirecting` is gone (if we're actually in the middle of a redirect, this component would be unloading/gone)

Reply:
There's a case where the target URL takes a long time to respond, which leaves this app visible.  It was important to inform the user that it's not our fault, that we are waiting for the other resource to respond.